contrib/src/main/java/com/datatorrent/contrib/parser/FixedWidthParser.java

```diff
@@ -0,0 +1,489 @@
+            } else if (value.compareToIgnoreCase(currentField.getFalseValue()) == 0) {
+              result = Boolean.parseBoolean("false");
+            } else {
+              throw new NumberFormatException();
+            }
+            break;
+          case DATE:
+            DateFormat df = new SimpleDateFormat(currentField.getDateFormat());
+            df.setLenient(false);
+            result = df.parse(value);
+            break;
+          default:
+            throw new RuntimeException("Invalid Type in Field", new Exception());
+        }
+        toEmit.put(fieldName,result);
+        if (typeInfo != null && pojoObject != null) {
+          typeInfo.setter.set(pojoObject, result);
+        }
+      } else {
+        toEmit.put(fieldName,value);
+      }
+    } catch (NumberFormatException e) {
+      throw new RuntimeException("Error parsing" + value + " to Integer type", e);
+    } catch (ParseException e) {
+      throw new RuntimeException("Error parsing" + value );
+    }catch (Exception e) {
+      throw new RuntimeException("Error setting " + value + " in the given class" + typeInfo.toString(), e);
+    }
+  }
+
+  /**
+   * Get the schema
+   *
+   * @return the Json schema
+   */
+  public String getJsonSchema()
+  {
+    return jsonSchema;
+  }
+
+  /**
+   * Set the jsonSchema
+   *
+   * @param jsonSchema schema to be set.
+   */
+  public void setJsonSchema(String jsonSchema)
+  {
+    this.jsonSchema = jsonSchema;
+  }
+
+  /**
+   * Get the header
+   *
+   * @return header- This will be string of field names, padded with padding character (if required)
+   */
+  public String getHeader()
+  {
+    return header;
+  }
+
+  /**
+   * Set the header
+   *
+   * @param header- This will be string of field names, padded with padding character (if required)
+   */
+  public void setHeader(String header)
+  {
+    this.header = header;
+  }
+
+  /**
+   * Get errorTupleCount
+   *
+   * @return errorTupleCount number of erroneous tuples.
+   */
+  @VisibleForTesting
+  public long getErrorTupleCount()
+  {
+    return errorTupleCount;
+  }
+
+  /**
+   * Get emittedObjectCount
+   *
+   * @return emittedObjectCount count of objects emitted.
+   */
+  @VisibleForTesting
+  public long getEmittedObjectCount()
+  {
+    return emittedObjectCount;
+  }
+
+  /**
+   * Get incomingTuplesCount
+   *
+   * @return incomingTuplesCount number of incoming tuples.
+   */
+  @VisibleForTesting
+  public long getIncomingTuplesCount()
+  {
+    return incomingTuplesCount;
+  }
+
+  /**
+   * Get parsedOutputCount
+   *
+   * @return parsedOutPutCount count of well parsed tuples.
+   */
+  @VisibleForTesting
+  public long getParsedOutputCount()
+  {
+    return parsedOutputCount;
+  }
+
+  /**
+   * Objects of this class represents a particular data member of the Class to be emitted.
+   * Each data member  has a name, type and a accessor(setter) function associated with it.
+   */
+  static class TypeInfo
```

Comment:
This class is used to ease the storage and mapping of setters. Something similar to what is being done in ProjectionOperator.